tools/harness-automation/autothreadharness/runner.py

```diff
@@ -280,6 +285,16 @@ def discover(names=None, pattern=['*.py'], skip='efp', dry_run=False, blacklist=
 
     suite.run(result)
 
+    if rerun_fails > 0:
```

Comment:
Is it possible to avoid recursive? Possibly, we could move this block to `main()` or involve another wrapper.

Reply:
I moved whole block to main()